client/src/components/ChatRoom.svelte

```diff
@@ -84,28 +104,50 @@
 </script>
 
 <div class="chatroom">
-  {#if countdown !== undefined}
+  {#if isStalker}
+    {#if countdown !== undefined}
+      <p class="countdown">{countdown}</p>
+    {/if}
+    {#if chatRoomInfo !== undefined}
+      <div>{firstPairInfo.displayName} MATCHED WITH {secondPairInfo.displayName}</div>
+    {/if}
+    <div class="messages">
+      {#each chatMessages as message}
+        {#if isFirstPair(message.sender)}
+          <p class="partner-name">{firstPairInfo.displayName}</p>
+          <p class="partner-msg">{message.text}</p>
+        {:else}
+          <p class="partner-name">{secondPairInfo.displayName}</p>
+          <p class="partner2-msg">{message.text}</p>
+        {/if}
+      {/each}
+    </div>
+  {:else}
     <p class="countdown">{countdown}</p>
```

Comment:
You can render this once since it doesn't matter if it is the stalker view or chatroom view.